### DISCUSSION TITLE
Failed to create collection Error: Unknown arguments: ['payload_schema']

### DISCUSSION THREAD

**Discussion by urbanscribe** [qdrant/qdrant-client#578] [General]
Running 1.8.2

Keep getting this error when trying to define the payload schema in a new collection

`    def create_collection(self):
        """
        Creates a collection with the specified vectors and payload configuration.

        Parameters:
            self: The object instance
            vectors_config: The configuration for the vectors
            payload_config: The configuration for the payload
        
        Returns:
            None
        """
        vectors_config = {
            "vectors": {
                "size": 768,  # The dimensionality of the vectors
                "distance": "Cosine"  # The distance metric for vector comparison
            }
        }
        payload_config = {
            "user_id": {"type": "integer"},
            "company_id": {"type": "integer"},
            "project_id": {"type": "integer"}
        }
        optimizers_config = {
            "max_optimization_threads": 2  # Adjust this value as needed
        }

        try:
            self.qdrant_client.create_collection(
                collection_name=self.collection_name,
                vectors_config=vectors_config,
                payload_schema=payload_config,
                optimizers_config=optimizers_config
            )
            
        except Exception as e:
            st.error(
                f"Failed to create collection '{self.collection_name}'. Error: {e}")`

Any thoughts?

**Comment by joein** [comment]
hey @urbanscribe 

`create_collection` does not have `payload_schema` argument
payload indices created separately via `create_payload_index`

you can find an example [here](https://qdrant.tech/documentation/concepts/payload/#payload-indexing)

**Comment by urbanscribe** [comment]
thanks, what confusing is you can fetch payload_schema from a CollectionInfo

 ```
  status: "CollectionStatus" = Field(..., description="Current statistics and configuration of the collection")
    optimizer_status: "OptimizersStatus" = Field(
        ..., description="Current statistics and configuration of the collection"
    )
    vectors_count: Optional[int] = Field(
        default=None,
        description="Approximate number of vectors in collection. All vectors in collection are available for querying. Calculated as `points_count x vectors_per_point`. Where `vectors_per_point` is a number of named vectors in schema.",
    )
    indexed_vectors_count: Optional[int] = Field(
        default=None,
        description="Approximate number of indexed vectors in the collection. Indexed vectors in large segments are faster to query, as it is stored in a specialized vector index.",
    )
    points_count: Optional[int] = Field(
        default=None,
        description="Approximate number of points (vectors + payloads) in collection. Each point could be accessed by unique id.",
    )
    segments_count: int = Field(
        ..., description="Number of segments in collection. Each segment has independent vector as payload indexes"
    )
    config: "CollectionConfig" = Field(..., description="Current statistics and configuration of the collection")
    payload_schema: Dict[str, "PayloadIndexInfo"] = Field(..., description="Types of stored payload")
```
so what are the best practices for managing the payload schema

perhaps there is a confusion on my part between setting a payload schema for a collection and creating an index inside that collection.  I am looking to do the former, I know how to do the latter.

**Reply by joein** [reply]
`payload_schema` in `CollectionInfo` shows payload indices created for that collection 

You don’t need (and currently can’t) set a payload schema to describe all the fields in your payload, if that is what you’re looking for

**Reply by urbanscribe** [reply]
Thanks. 

Does that mean every time I upsert a file with any new payload/metadata field/value it will accept it and thus defining a schema for validation ahead of time is not necessary or supported?

Payload items, schema and indexing are used a little interchangeably throughout the docs. 

Thanks in advance.

**Comment by joein** [comment]
Yes, it is correct

> Payload items, schema and indexing are used a little interchangeably throughout the docs.

Could you provide some links which causes confusion please?
